posthog/temporal/tests/batch_exports/test_http_batch_export_workflow.py

```diff
@@ -0,0 +1,466 @@
+            await generate_test_events_in_clickhouse(
+                client=clickhouse_client,
+                team_id=team_id,
+                start_time=data_interval_start,
+                end_time=data_interval_end,
+                count=5,
+                count_outside_range=0,
+                count_other_team=0,
+                event_name=event_name,
+            )
+
+    insert_inputs = HttpInsertInputs(
+        team_id=team_id,
+        data_interval_start=data_interval_start.isoformat(),
+        data_interval_end=data_interval_end.isoformat(),
+        exclude_events=exclude_events,
+        batch_export_schema=None,
+        **http_config,
+    )
+
+    mock_server = MockServer()
+    with aioresponses(passthrough=[settings.CLICKHOUSE_HTTP_URL]) as m, override_settings(
+        BATCH_EXPORT_HTTP_UPLOAD_CHUNK_SIZE_BYTES=5 * 1024**2
+    ):
+        m.post(TEST_URL, status=200, callback=mock_server.post, repeat=True)
+        await activity_environment.run(insert_into_http_activity, insert_inputs)
+
+    await assert_clickhouse_records_in_mock_server(
+        mock_server=mock_server,
+        clickhouse_client=clickhouse_client,
+        team_id=team_id,
+        data_interval_start=data_interval_start,
+        data_interval_end=data_interval_end,
+        exclude_events=exclude_events,
+    )
+
+
+async def test_insert_into_http_activity_throws_on_bad_http_status(
+    clickhouse_client, activity_environment, http_config, exclude_events
+):
+    """Test that the insert_into_http_activity function throws on status >= 400"""
+    data_interval_start = dt.datetime(2023, 4, 20, 14, 0, 0, tzinfo=dt.timezone.utc)
+    data_interval_end = dt.datetime(2023, 4, 25, 15, 0, 0, tzinfo=dt.timezone.utc)
+
+    # Generate a random team id integer. There's still a chance of a collision,
+    # but it's very small.
+    team_id = randint(1, 1000000)
+
+    await generate_test_events_in_clickhouse(
+        client=clickhouse_client,
+        team_id=team_id,
+        start_time=data_interval_start,
+        end_time=data_interval_end,
+        count=1,
+        count_outside_range=10,
+        count_other_team=10,
+        duplicate=True,
+        properties={"$browser": "Chrome", "$os": "Mac OS X"},
+        person_properties={"utm_medium": "referral", "$initial_os": "Linux"},
+    )
+
+    insert_inputs = HttpInsertInputs(
+        team_id=team_id,
+        data_interval_start=data_interval_start.isoformat(),
+        data_interval_end=data_interval_end.isoformat(),
+        exclude_events=exclude_events,
+        batch_export_schema=None,
+        **http_config,
+    )
+
+    with aioresponses(passthrough=[settings.CLICKHOUSE_HTTP_URL]) as m, override_settings(
+        BATCH_EXPORT_HTTP_UPLOAD_CHUNK_SIZE_BYTES=5 * 1024**2
+    ):
+        m.post(TEST_URL, status=400, repeat=True)
+        with pytest.raises(ClientResponseError):
+            await activity_environment.run(insert_into_http_activity, insert_inputs)
```

Comment:
Even if the activity throws, if the exception is not in `non_retryable_error_types` the workflow will retry the activity.